internal/osutil/file.go

```diff
@@ -94,3 +96,27 @@ func IsRegularFile(path string) (bool, error) {
 
 	return fileStat.Mode().IsRegular(), nil
 }
+
+// DetectFileType returns a file's content type given path
+func DetectFileType(path string) (string, error) {
+	rc, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer rc.Close()
+	var header [512]byte
+	_, err = io.ReadFull(rc, header[:])
```

Comment:
It is better to have a `LimitedReader` instead of a 512-byte array as a file may be smaller than 512 bytes